apple/swift-docc-render
# Allow toggling sidebar navigator on/off, for large breakpoints

Bug/issue #, if applicable: 89564104

## Summary

Allows toggling the navigator ON/OFF, for large resolutions. This is done through the Nav toggle icon or by dragging.
Users can drag the navigator all the way to the minimum allowed size. Once they pass 2x the minimum, the sidebar gets hidden. 

The user can expand it back out, if they have not let go of the mouse button upon dragging OR by clicking on the nav toggle. (should work similar to Mail.app)

Hidden state is persisted in localStorage, so it wont re-open for users who have closed it once.

## Dependencies

NA

## Testing

Steps:
1. Try to click on the toggle to open/close the navigator on Large screen sizes. Assert nothing bleeds out when hidden.
2. Make sure tabbing across the page does not focus anything inside the navigator
3. Refresh the page to assert its kept closed.
4. Open the navigator and start dragging all the way until it stops. Now drag 2x the left space and assert its toggled off.
5. Without releasing your drag, try to expand out again. It should snap to the smallest possible size again.
6. If you drag until closed, it should persist in LS the closed state, and later allow you to toggle open again.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
